client/context/context.jsx

```diff
@@ -117,6 +119,21 @@ export const AppProvider = ({ children }) => {
     }
   }
 
+  // Get user book club by id
+  const getUserClubsById = async (ids) => {
+    const userClubData = [];
+    try {
+      ids.map(async (clubId) => {
```

Comment:
This does each request one after another instead of sending them all out at once. Could be a significant area of optimization. 

We can worry about this later